### DISCUSSION TITLE
RNA_STAR_route define the group to compare to a control

### DISCUSSION THREAD

**Discussion by Miladebrahim** [igordot/sns#20] [Q&A] [answered]
Hi Igor,

How can i define the group that i need to compare to a control (i.e. if i assign the samples as Xmutant and Xwildtype, the results come back comparing Xwildtype to Xmutant) how can i reverse this ? so that i compare Xmutant to Xwildtype ? 

Thank you

**Comment by igordot** [comment, marked as answer]
The order of groups in `samples.groups.csv` should define the comparisons. If there are two groups, the first one is the reference.

**Comment by Miladebrahim** [comment]
That is excellent
Thank you

**Comment by anke-kloock** [comment]
Hi Igor, I am using your pipeline and find it super helpful! Thank you so much! I have successfully run the rna-star pipeline and wanted to run the rna-star-groups-dge. Somehow I am however having issues for the pipeline to accept the file "samples.groups.csv". Even though I have renamed the groups and replaced it in the same folder with the same name, I keep getting the error message of "ExitCode 1" (via email). In the terminal I get this error message: CMD: sbatch --time=10:00:00 --nodes=1 --ntasks=1 --cpus-per-task=5 --mem=50G --job-name=sns.rna-star-groups-dge --mail-user=klooca01@nyulangone.org --mail-type=FAIL,REQUEUE --export=NONE --wrap='bash /gpfs/data/jhubbardlab/sns/routes/rna-star-groups-dge.sh /gpfs/data/jhubbardlab' 
Submitted batch job 48128185

How do I write the file "samples.groups.csv" properly into the repository, or is there anything else that I do not do correctly and need to adjust? 

Thanks so much for your help!

**Comment by igordot** [comment]
Did you check the full log file?

In the `logs-sbatch` directory, there is a log file for every sbatch job. You can view it in the terminal with `cat logs-sbatch/slurm-[jobid].out` or download it and open it with any text editor. If you are not sure what to look for, you can copy and paste the contents or attach the whole file here. Generally, the last few lines (right before the job ends) are the most relevant ones for troubleshooting the error.

**Comment by anke-kloock** [comment]
Thanks for the fast reply Igor! There was no logs-sbatch directory prior to running the command you sent. And after the command it says "No such file or directory". After your command there is a folder with logs-sbatch, but its empty.

**Comment by anke-kloock** [comment]
I found the file you were referring to, even though it was not in the expected folder. I think the issue is that I do not have replicates, which are required since version v1.22. Is there any chance I could use an older version?

**Comment by igordot** [comment]
It is not possible to estimate variability of a gene without replicates, so any statistics would not be particularly meaningful.

If you don't have replicates, you can use DESeq2 1.18. It's part of Bioconductor 3.6, which requires R 3.4. Many other packages would need to be downgraded as well. As you may imagine, that would require a lot of troubleshooting.

Alternatively, you can assign all samples to a single group and DESeq2 will generate a table of normalized counts. You can then calculate a fold changes based on that.

**Comment by anke-kloock** [comment]
Hi Igor, thanks for coming back to me so quickly! I have pooled all my worm replicates into one sample. I guess I should not have done that. 

I am now left with three mutants, and no replicates. so if I were to assign them a single group, I would not be able to make comparisons? 

Alternatively, is there a way of splitting the data into three batches now? I guess that is not the proper way to do, but I am trying to avoid having to run the experiment again. 

Thank you so much for your help in figuring this out!

**Comment by igordot** [comment]
I assume you mean you pooled them before library prep and sequencing. It is too late in that case.

You could still calculate fold changes and look at the top genes. You just can't do any proper stats.
